### DISCUSSION TITLE
设备的实体ID在哪获取

### DISCUSSION THREAD

**Discussion by fengyanbiao** [XiaoMi/ha_xiaomi_home#492] [General]
你好，已知设备ID，但是相关的实体ID应该怎么拼接，规范在哪找

**Comment by fengyanbiao** [comment]
![米家](https://github.com/user-attachments/assets/4e22225d-7b27-44e7-8ead-ade534878d53)

**Comment by fengyanbiao** [comment]
![PixPin_2024-12-29_12-56-20](https://github.com/user-attachments/assets/6903d0f1-21a9-4a1c-afe4-a26aa7f40f9c)
美的的实体ID规则是这样的，咱们小米的实体规则是怎么样的呀？有没有文档

**Comment by zghnwsq** [comment]
did应该是小米分配的

**Reply by fengyanbiao** [reply]
是的，但是小米好像没有把id的生成规则开发出来

**Comment by zghnwsq** [comment]
翻一下代码吧，在miot_device.py  
gen_device_entity_id

```python
    def gen_device_entity_id(self, ha_domain: str) -> str:
        return (
            f'{ha_domain}.{self._model_strs[0][:9]}_{self.did_tag}_'
            f'{self._model_strs[-1][:20]}')
```

**Reply by JimmyKmi** [reply]
正解，翻一下代码就知道了

**Reply by fengyanbiao** [reply]
这个看不出来，我们自己有个物联网平台，然后相关http请求去调用home-assistant的接口来控制设备，所以必须要知道entity_id的规则
